libr/util/sys.c

```diff
@@ -1199,10 +1199,32 @@ R_API char **r_sys_get_environ (void) {
 	if (!env) {
 		env = r_lib_dl_sym (NULL, "environ");
 	}
+
+	// save original environ for debugging
+	if (env) {
+		size_t count, i;
+
+		for (i = 0; env[i]; i++)
+			;
+		count = i;
+
+		origenv = malloc (sizeof (char*) * (count + 1));
```

Comment:
use calloc and avoid the multiplication. Also this global allocation is never free'd and this can be a problem as long as nothing ensures this is called only once and there's no way to deallocate it.

Also, env is writable. and its linearly allocated in memory, we can just do a single allocation for the whole buffer and then another allocation for the pointer storage. instead of having one strdup per-env. But this can result on some other problems we probably dont want to handle.

What are your thoughs on this @ret2libc ?

Reply:
Use `R_NEWS`

Reply:
Although in theory what @trufae suggested is probably the best thing memory-wise, I would stick with this simpler solution for the beginning, unless we realize those additional strdup are really a problem.